core/arch/arm32/include/kernel/misc.h

```diff
@@ -29,11 +29,24 @@
 #define KERNEL_MISC_H
 
 #include <types_ext.h>
+#include <utee_types.h>
+#include <arm32.h>
 
 size_t get_core_pos(void);
 
 uint32_t read_usr_sp(void);
 uint32_t read_usr_lr(void);
 
+static inline int get_generic_timer_info(struct generic_timer_info *timer_info)
```

Comment:
The design is bad, this should be separated into architecture independent part and architecture dependent part.